### PR TITLE
Fix authenticated middleware schema check for users who cant access the local cluster

### LINKED ISSUE #9336: Standard users are unable to view workloads for a downstream cluster
**Setup**
- Rancher version: v2.7-head (37ef305)
- Rancher UI Extensions: NA
- Browser type & version: Firefox 116.0b3 (64-bit)

**Describe the bug**

Attempting to list or navigate to a workload as a standard user, results in a `Error Resource type pod not found` message. 

**To Reproduce**

1. Log into Rancher Dashboard as the Admin user
2. Navigate to Cluster Management => Create
3. Create a new RKE2 Cluster with Digital Ocean as the cluster provider, accept all defaults
4. Navigate to Users & Authentication => Create
5. Create a new user with "Standard User" selected under Global Permissions
6. Navigate to Explore Cluster & select your newly created cluster from Step 3
7. Navigate to Workloads => Pods
8. Create a simple pod with `nginx:latest` as the image, give it a name and accept all defaults
9. Navigate to Cluster & Project Members => Project Membership
10. Click on "Add" for the Default group
11. Add the user created in Step 5 as a Member
12. Log into Rancher Dashboard as the user created in Step 5 
13. Navigate to Explore Cluster and select the cluster created in Step 3
14. Select Workloads => Pods

**Result**

We are directed to the error page with a message that reads `Error Resource type pod not found` 

**Expected Result**

A list of Pods is displayed, including the pod created in Step 8.

**Screenshots**

image

**Additional context**

Navigating to Workloads as the user created in Step 5 will successfully list all of the workloads. Attempting to select on of these workloads will produce the error.

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
### Summary
Fixes #9336

### Occurred changes and/or fixed issues
This PR updates the schema check in auth middleware to also look in the current store. The problem described in #9336 seems to be happening because the standard user can't see pods in the management store, as they only have access to pods in the downstream cluster.

### Areas or cases that should be tested
Pages generated by dynamic routes in downstream clusters as a user with limited access to the local cluster; bogus routes should still fail

### CODE REVIEW

File: cypress/e2e/tests/navigation/not-found-page.spec.ts [hunk 1]
```diff
@@ -44,4 +44,12 @@ describe('Not found page display', () => {
     notFound.errorTitle().contains('Error');
     notFound.errorMessage().contains('Product bogus-product-id not found');
   });
+
+  it('Will not show a 404 if we have a valid product + resource', () => {
```
Comment (richard-cox) [inline cypress/e2e/tests/navigation/not-found-page.spec.ts:48]: Comment only. Super nice addition!

File: cypress/e2e/tests/navigation/not-found-page.spec.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import NotFoundPagePo from '@/cypress/e2e/po/pages/not-found-page.po';
 
-describe('Not found page display', () => {
+describe('Not found page display', { tags: ['@adminUser', '@standardUser'] }, () => {
```
Comment (richard-cox) [inline cypress/e2e/tests/navigation/not-found-page.spec.ts:3]: Comment only. Cripes none of these were running? We should have an issue to error if there are e2e jobs that aren't run

**Review by aalves08 — COMMENTED** [review]
will wait for changes from @richard-cox (we had a sync)

**Review by richard-cox — DISMISSED** [review]
I've done a deep dive and pushed an update

```
- We don't need to check management and rancher stores, just the current store
- We need to ensure that the product has been correctly set to get it's current store
- Also check that schemaFor exists, to cover extension cases
- replace workaround for workload type with a check for virtual types
  - note - spoofed types have schemas
- add additional e2e tests for other troublesome areas
```

I've also added another e2e test to check for navigates to a page (as well as the existing load on a page). This will catch scenarios with the new code and product hasn't been set correctly.

@aalves08 @mantis-toboggan-md if you're both happy lets merge

**Comment by mantis-toboggan-md** [comment]
These looks like solid cahgnes to me @richard-cox - I'm good with merging this if @aalves08 is

File: cypress/e2e/tests/navigation/not-found-page.spec.ts (hunk 2 above)
Comment (mantis-toboggan-md) [reply cypress/e2e/tests/navigation/not-found-page.spec.ts:8]: I've written an issue for this now: https://github.com/rancher/dashboard/issues/9350

**Review by aalves08 — APPROVED** [review]
lgtm @mantis-toboggan-md ! :)

**Comment by mantis-toboggan-md** [comment]
e2e tests were failing for the standard user, who doesn't have access to the local cluster. I've updated them to check resources both the standard user and admin should have access to, except the workload tests, which now run only for admins. 

I also noticed that trying to nav to `c/_/manager/provisioning.cattle.io.cluster/fleet-default/bogus-id` as a standard user returns a 403 instead of 404...do we want to catch this in the same way?

**Review by richard-cox — APPROVED** [review]
I'm really keen to get the e2e tests working with the standard user. That revolves around setting up the standard user with access permissions to the local cluster and workload resources. Will do this in another PR though, as this fix is important to get in

Edit: https://github.com/rancher/dashboard/pull/9360

**Pull request merged**